auth0_flutter/lib/src/web/js_interop.dart

```diff
@@ -1,2 +1,86 @@
 // Here go the annotations –from the js package– that allow calling JS code
-// See https://pub.dev/packages/js#usage
\ No newline at end of file
+// See https://pub.dev/packages/js#usage
+
+// ignore_for_file: non_constant_identifier_names
+
+@JS('auth0')
+library auth0;
+
+import 'dart:js_util';
+
+import 'package:js/js.dart';
+
+@JS()
+@staticInterop
+@anonymous
+abstract class AuthorizationParams {
+  external factory AuthorizationParams(
+      {final String? audience, final String? redirect_uri});
+}
+
+extension on AuthorizationParams {
+  external String? audience;
+  external String? redirect_uri;
+
+  // https://github.com/dart-lang/sdk/issues/38445
+  // Workaround of setting property names doesn't seem to work. Tried:
+  // @JS('redirect_uri')
+  // external String? get redirectUri;
+
+  // @JS('redirect_uri')
+  // external set redirectUri(final String? redirectUri);
+}
+
+/// Rebuilds the input object, omitting values that are null
+@JS()
+T stripNulls<T extends Object>(final T obj) {
+  final keys = objectKeys(obj);
+  final output = newObject<Object>();
+
+  for (var i = 0; i < keys.length; i++) {
+    final key = keys[i] as String;
+    final value = getProperty(obj, key) as dynamic;
+
+    if (value != null) {
+      setProperty(output, key, value);
+    }
+  }
+
+  return output as T;
+}
```

Comment:
Some mapping between the keys to strip out values that were `null` is necessary, otherwise the nulls get sent through the URL to the IdP, which is undesirable. The SPA SDK doesn't strip these out.

This is needed as Dart JS interop treats `null` and `undefined` as completely identical. Consider this example:

```dart
final String? redirectUri = null;

client.loginWithRedirect(RedirectLoginOptions(
  authorizationParams: AuthorizationParams(
    redirect_uri: redirectUri)));

// redirect_uri is `null` in the URL
```

Without some mapping function that executes in JS land, an alternative would be to do two different calls depending on whether `redirectUri` is null or not, which doesn't scale as this applies to all optional parameters.

Or, use `Map<string, string>` in the API, in which case we lose strong typing but also in JS land this equates to using a Dart Map type, which is different from a plain JS object, so the JS interop API doesn't line up with reality (**must** use classes annotated with `@anonymous`).

Reply:
Again, not precious on the name of the function, happy to hear suggestions.